libyul/YulStack.cpp

```diff
@@ -307,22 +362,31 @@ std::string YulStack::print(
 	CharStreamProvider const* _soliditySourceProvider
 ) const
 {
+	yulAssert(m_stackState >= Parsed);
 	yulAssert(m_parserResult, "");
 	yulAssert(m_parserResult->code, "");
 	return m_parserResult->toString(&languageToDialect(m_language, m_evmVersion), m_debugInfoSelection, _soliditySourceProvider) + "\n";
 }
 
 Json YulStack::astJson() const
 {
+	yulAssert(m_stackState >= Parsed);
 	yulAssert(m_parserResult, "");
 	yulAssert(m_parserResult->code, "");
 	return  m_parserResult->toJson();
 }
 
 std::shared_ptr<Object> YulStack::parserResult() const
 {
-	yulAssert(m_analysisSuccessful, "Analysis was not successful.");
+	yulAssert(m_stackState >= Parsed, "Parsing was not successful");
+	yulAssert(m_stackState >= AnalysisSuccessful, "Analysis was not successful.");
```

Comment:
These look redundant:
```suggestion
	yulAssert(m_stackState >= AnalysisSuccessful, "Analysis was not successful.");
```